specification/dartLangSpec.tex

```diff
@@ -619,26 +621,48 @@ \subsection{Scoping}
 \LMLabel{scoping}
 
 \LMHash{}%
-A \Index{namespace} is a mapping of names denoting declarations to actual declarations.
-Let $NS$ be a namespace.
-We say that a name $n$ \Index{is in} $NS$ if $n$ is a key of $NS$.
-We say a declaration $d$ \NoIndex{is in} $NS$ if a key of $NS$ maps to $d$.
+A \Index{namespace} is a mapping from names to declarations.
+It may be considered as a partial function.
+It may also be seen as a set of bindings from a name to a declaration,
+$n\mapsto{}D$,
+with the constraint that if a namespace has both
+$n\mapsto{}D_1$ and $n\mapsto{}D_2$
+then $D_1$ and $D_2$ is the same declaration.
```

Comment:
Your description already implies that two separate pieces of source code are two distinct declarations. Even if they are textually identical (and located in the same binding environment), we will consider that to be a name clash and report an error (rather than considering them as instructions to add an element to a set, and just adding one element because those two declarations denote the same semantic "value"). So I don't think we have a real ambiguity here.

Do you wish to pursue an explicit discussion about this topic in the specification? I'll leave the text unchanged for now, and then we can handle it separately.

Reply:
> the initial "both" is suggesting that they were different elements.

I think this is a clash between a mathematical style and an assumption which is reasonable in non-mathematical language ("we don't give two different names to the same thing"). I adjusted the text to reduce the potential for misunderstandings.